ArjunaCore/arjuna/classes/com/arjuna/ats/arjuna/coordinator/TransactionReaper.java

```diff
@@ -108,7 +108,16 @@ public final void check()
                 // TODO close window where first can change - maybe record nextDynamicCheckTime before probing first,
                 // then use compareAndSet? Although something will need to check before sleeping anyhow...
                 if (reaperElement == null) {
-                    nextDynamicCheckTime.set(Long.MAX_VALUE);
+                    if (!_inShutdown)
+                        nextDynamicCheckTime.set(Long.MAX_VALUE);
+                    else {
+                        // If I am here during shutdown it means that _inShutdown was read from a cached value
+                        // during removeElementClient
+                        nextDynamicCheckTime.set(0);
```

Comment:
@jmfinelli do you have to do this line? IIUC this is intended to be when removeElementClient was called and returns a stale _inShutdown so with this change I think you are expecting that the thread trying to terminate the reaper must be stuck at https://github.com/jbosstm/narayana/blob/main/ArjunaCore/arjuna/classes/com/arjuna/ats/arjuna/coordinator/TransactionReaper.java#L735 so is nextDynamicCheckTime needing to be zero to clear something else up?

Reply:
I used `nextDynamicCheckTime.set(0)` as I thought I needed to let the Reaper Thread know that it doesn't need to sleep again. I realised now that this is not needed as `checkInPeriod` will return a negative value which will not send the Reaper Thread to sleep. Thanks for asking this question